source/slides/02_linux_basics.rst

```diff
@@ -0,0 +1,507 @@
+load new commands into your editor while you were editing. You could edit the
+programs you were using and then go on editing with them.
+
+ -- Richard Stallman, http://www.gnu.org/gnu/rms-lisp.html
+
+Vim
+---
+
+.. figure:: ../_static/vim_logo.jpeg
+    :align: right
+
+.. note::
+
+    originally written for Amiga systems (Commodore PCs), 1988
+    vim released 1991
+    vimscript, Lua (as of Vim 7.3), Perl, Python, Racket, Ruby, Tcl (tool
+    command language).
+    vi written by Bill Joy in 1976, visual mode for line editor called ex
+    line editors are from age of teleprinters, no cursors
+
+* Available almost everywhere
+* Lightweight
+* Design decisions explained in http://docs.freebsd.org/44doc/usd/12.vi/paper.html
+* Modal editor (command, insert, visual)
+
+How to choose
+-------------
+
+* What can the people around you help with?
+* Try both; choose one and get good at it
+* Have a good answer when people ask why you made that choice
+    * "Because it's familiar" is tolerated
+    * "Because I was initially taught it" is common but accepted (honesty)
+    * "Because ``$usecase``" provokes argument but more respected
+    * "Because I tried both and picked this one" is rare but good
+* Your use case as a sysadmin or developer
+
+Modes
+-----
+
+.. figure:: ../_static/vim_modes.png
+    :align: center
+    :scale: 75%
+
+How to tell?
+
+.. code-block:: bash
+
+    -- INSERT --                                          144,1    36%
+    -- VISUAL --                                          144,77   36%
+
+Commands
+--------
+
+.. note::
+    Moving around in a file
+    Search / replace
+    Text manipulation, ie: cw, dw, c$, yy / p, x, .
+
+.. figure:: ../_static/vim_cheatsheet.gif
+    :scale: 75%
+
+Moving Around
+-------------
+
+::
+
+    h move one character to the left.
+    j move down one line.
+    k move up one line.
+    l move one character to the right.
+    0 move to the beginning of the line.
+    $ move to the end of the line.
+    w move forward one word.
+    b move backward one word.
+    G move to the end of the file.
+    gg move to the beginning of the file.
+    . move to the last edit.
+
+Configuration / customization
+-----------------------------
+
+.. note:: there are many many options and pre-existing packages to make
+    editing nice for sysadmins and developers
+
+* ``.vimrc``
+* ``:set``
+
+Some sets of Vim plugins and configurations are available
+
+* https://github.com/astrails/dotvim
+* https://github.com/carlhuda/janus
+
+Use them for research on what's available to improve dev productivity
+
+Learning Resources
+------------------
+
+* ``$ vimtutor``
+* http://vim-adventures.com/
+
+.. figure:: ../_static/learning_curves.jpg
+    :align: center
+    :scale: 140%
+
+Regular expressions
+-------------------
+
+You should know basic substitution:
+
+::
+
+    :%s/foo/bar/g
+
+On IRC, Hamper does rudimentary regex in the form ``s/foo/bar/`` applying only
```

Comment:
Do we want to mention the Hamper irc bot at all here or generalize this statement a bit?